src/main/java/de/uni_tuebingen/ub/nppm/servlet/AjaxServlet.java

```diff
@@ -35,6 +45,240 @@ protected void autocomplete(HttpServletRequest request, HttpServletResponse resp
+                    String language = Language.getLanguage(request);
+                    StringBuilder sb = new StringBuilder();
+                    int save_namenkommentarID = -1;
+
+                    for (Map.Entry<Integer, List<Integer>> entry : namenkommentarIdToEinzelbelegIdsMap.entrySet()) {
+                        int namenkommentarID = entry.getKey();
+                        List<Integer> einzelbelegIDs = entry.getValue();
+                        String plemma = namenkommentarIdToPlemmaMap.get(namenkommentarID);
+
+                        String einzelbelegIDsString = einzelbelegIDs.toString();
+
+                        String s = DatenbankDB.getLabel(language, "einzelbeleg", "foundZusatznamenkommentar");
+
+                        String result = String.format(s, belegform, plemma, String.valueOf(namenkommentarID), einzelbelegIDsString);
+
+                        sb.append(result).append("\n\n");
+
+                        if (namenkommentarIdToEinzelbelegIdsMap.size() == 1) {
+                            save_namenkommentarID = namenkommentarID;
+                        }
+                    }
+
+                    if (namenkommentarIdToEinzelbelegIdsMap.size() == 1) {
+                        sb.append(DatenbankDB.getLabel(language, "einzelbeleg", "replaceZusatznamenkommentar"));
+                    } else {
+                        sb.append(DatenbankDB.getLabel(language, "einzelbeleg", "cleanData"));
+                    }
+
+                    outputList.add(sb.toString());
+
+                    response.setContentType("application/json; charset=UTF-8");
+
+                    JSONObject jsonObject = new JSONObject();
+                    jsonObject.put("EinzelbelegID", EinzelbelegID);
+                    jsonObject.put("outputListZ", outputList);
+
+                    if (save_namenkommentarID != -1) {
+                        jsonObject.put("namenkommentarID", save_namenkommentarID);
+                    }
+
+                    response.getWriter().print(jsonObject.toString());
+                } else {
+                    response.setStatus(HttpServletResponse.SC_NO_CONTENT);
+                }
+            }
+        } catch (Exception e) {
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    protected void confirmZusatzNamenKommentar(HttpServletRequest request, HttpServletResponse response) {
+        try {
+            String EinzelbelegID = request.getParameter("EinzelbelegID");
+            String namenkommentarID = request.getParameter("namenkommentarID");
+
+            if (EinzelbelegID == null || namenkommentarID == null) {
+                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            } else {
+                // Logik zum Speichern des Zusatznamen-Kommentars
+                EinzelbelegDB.insertNamenkommentar(EinzelbelegID, namenkommentarID);
+
+                response.setContentType("application/json; charset=UTF-8");
```

Comment:
In this function it is not necessary to write an application/json result, since you're just returning an empty object here. It is enough for the caller on the JS side to use the HTTP response code to check whether the operation was successful or not